src/proxy/compute/data_collection/components/convert.rs

```diff
@@ -134,8 +135,21 @@ impl From<payload::Session> for provider::Session {
     }
 }
 
-fn convert_dict<T: ToString>(dict: HashMap<String, T>) -> Vec<(String, String)> {
-    dict.into_iter()
-        .map(|(key, value)| (key, value.to_string()))
-        .collect()
+fn convert_dict(dict: Option<HashMap<String, serde_json::Value>>) -> Vec<(String, String)> {
+    let mut vec: Vec<(String, String)> = vec![];
+    if dict.is_none() {
+        return vec![];
+    }
+    for (k, v) in dict.unwrap().iter() {
+        if v.is_object() || v.is_array() {
+            continue;
+        }
+        let s = v.as_str();
+        if let Some(s) = s {
+            vec.push((k.clone(), s.to_string()));
+        } else {
+            vec.push((k.clone(), v.to_string()));
+        }
+    }
+    vec
```

Comment:
You can use the `let Some($pat) = $value else { $code };` pattern to avoid unwrapping around (both generate kinda the same assembly at the end but the "let ... else ..." pattern is a really nice syntax to benefit the full type-safety of Rust :slightly_smiling_face:)

Also i would suggest using the iterator-based way of converting stuff using filter and map, but that's mainly a "procerural vs functional" styling issue :smile: 

```suggestion
    use serde_json::Value;

    let Some(dict) = dict else {
        return Vec::new();
    };

    dict.into_iter()
        .filter(|(_, value)| !(value.is_array() || value.is_object()))
        .map(|(k, v)| {
            let value = if let Value::String(ref v) = v {
                v.clone()
            } else {
                v.to_string()
            };

            (k, value)
        })
        .collect()
```